defaults/pythonSubmitDefault.py

```diff
@@ -498,234 +498,92 @@ def listCommands(self):
-    Options not to be included in the command line should be set to pythons None (except booleans, which should be set to False)
-
-    Parameters
-    -----------
-    programOptions : pythonProgramOptions
-        Contains program options
-
-    Returns
-    --------
-    commands : str or list of strs
-    """
-    booleanChoices = programOptions.booleanChoices()
-    booleanCommands = programOptions.booleanCommands()
-
-    numericalChoices = programOptions.numericalChoices()
-    numericalCommands = programOptions.numericalCommands()
-
-    stringChoices = programOptions.stringChoices()
-    stringCommands = programOptions.stringCommands()
-
-    listChoices = programOptions.listChoices()
-    listCommands = programOptions.listCommands()
-
-    if programOptions.hyperparameterGrid == True:
-        command = hyperparameterGridCommand(programOptions.compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands,programOptions.shareSeeds)
-    elif programOptions.hyperparameterList == True:
-        if programOptions.hyperparameterGrid == True:
-            raise ValueError("You can't have both a list and a grid!")
-        command = hyperparameterListCommand(programOptions.compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands,programOptions.shareSeeds)
-    else:
-        command = [generateCommandLineOptions(programOptions.compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands)]
-
-    #command = [generateCommandLineOptions(programOptions.compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands)]
-
-    return command
-
-def generateCommandLineOptions(compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands):
-
-    nBoolean = len(booleanChoices)
-    assert len(booleanCommands) == nBoolean
-
-    nNumerical = len(numericalChoices)
-    assert len(numericalCommands) == nNumerical
-
-    nString = len(stringChoices)
-    assert len(stringCommands) == nString
-
-    nList = len(listChoices)
-    assert len(listCommands) == nList
-
-    command = compas_executable + ' '
-
-    for i in range(nBoolean):
-
-        if booleanChoices[i] == True:
-
-            command += booleanCommands[i] + ' '
-
-    for i in range(nNumerical):
-
-        if not numericalChoices[i] == None:
-
-            command += numericalCommands[i] + ' ' + str(numericalChoices[i]) + ' '
-
-    for i in range(nString):
-
-        if not stringChoices[i] == None:
-
-            command += stringCommands[i] + ' ' + stringChoices[i] + ' '
-
-    for i in range(nList):
-
-        if listChoices[i]:
+    def generateCommandLineOptionsDict(self):
```

Comment:
nothing particularly bad, but just wondering what changed the indentation? If it works, it works!

Reply:
By redefining the function as a method of the pythonProgramOptions class, it made it easier to import only the class into the stroopwafel interface.